mage/Magefile.go

```diff
@@ -1114,7 +1114,26 @@ STANDALONE=0
 		return fmt.Errorf("failed to create out/logs directory: %w", err)
 	}
 
-	if err := sh.RunV(filepath.Join("scripts", "update_provider_defaultos.sh"), "microvisor"); err != nil {
+	// Check for DEFAULT_OS_PROFILE environment variable, default to "microvisor" if not set
+	defaultOS := os.Getenv("DEFAULT_OS_PROFILE")
+	if defaultOS == "" {
+		defaultOS = "microvisor"
+	}
+
+	// Validate DEFAULT_OS_PROFILE value
+	validOSProfiles := []string{"ubuntu", "microvisor", "microvisor-standalone"}
+	isValid := false
+	for _, validProfile := range validOSProfiles {
+		if defaultOS == validProfile {
+			isValid = true
+			break
+		}
+	}
+	if !isValid {
+		return fmt.Errorf("invalid DEFAULT_OS_PROFILE value: %s. Valid options are: %s", defaultOS, strings.Join(validOSProfiles, ", "))
+	}
```

Comment:
```suggestion
	// Validate DEFAULT_OS_PROFILE value
	validOSProfiles := []string{"ubuntu", "microvisor", "microvisor-standalone"}
	if !slices.Contains(validOSProfiles, defaultOS) {
		return fmt.Errorf("invalid DEFAULT_OS_PROFILE value: %s. Valid options are: %s", defaultOS, strings.Join(validOSProfiles, ", "))
	}
```

I think this can be made shorter